costmap_2d/src/costmap_2d.cpp

```diff
@@ -219,6 +224,23 @@ bool Costmap2D::worldToMap(double wx, double wy, unsigned int& mx, unsigned int&
   return false;
 }
 
+bool Costmap2D::worldToMapContinuous(double wx, double wy, float& mx, float& my) const
+{
+  if (wx < origin_x_ || wy < origin_y_)
+  {
+    return false;
+  }
+
+  mx = static_cast<float>((wx - origin_x_) / resolution_) + 0.5f;
+  my = static_cast<float>((wy - origin_y_) / resolution_) + 0.5f;
```

Comment:
Why `+ 0.5f` ?

Reply:
To use the center of the cell, instead of the border. 
disclaimer: It's not my choice,,, I copied from nav2

Reply:
please write your reply above to the code as a comment. 
people will start :thinking: in 2026 again.